File: scripts/resize_images.js
const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

const src = path.join(__dirname, '..', 'assets', 'hero-bg.jpg');
const outDir = path.join(__dirname, '..', 'assets');
const sizes = [480, 768, 1200];

(async () => {
  try {
    if (!fs.existsSync(src)) {
      console.error('Source image not found:', src);
      process.exit(1);
    }

    for (const w of sizes) {
      const outPath = path.join(outDir, `hero-bg-${w}.jpg`);
      await sharp(src)
        .resize({ width: w })
        .jpeg({ quality: 85, mozjpeg: true })
        .toFile(outPath);
      console.log('Wrote', outPath);
    }

    console.log('Done');
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
})();

File: scripts/resize_images.py
from PIL import Image
import os

src = os.path.join(os.path.dirname(__file__), '..', 'assets', 'hero-bg.jpg')
out_dir = os.path.join(os.path.dirname(__file__), '..', 'assets')

sizes = [480, 768, 1200]

if not os.path.exists(src):
    print('Source image not found:', src)
    raise SystemExit(1)

img = Image.open(src)
for w in sizes:
    ratio = w / float(img.width)
    h = int(img.height * ratio)
    resized = img.resize((w, h), Image.LANCZOS)
    out_path = os.path.join(out_dir, f'hero-bg-{w}.jpg')
    resized.save(out_path, format='JPEG', quality=85, optimize=True)
    print('Wrote', out_path)
print('Done')
